plugins/@grouparoo/pipedrive/__tests__/export/export-profile.ts

```diff
@@ -0,0 +1,410 @@
+      oldGroups: [],
+      newGroups: [groupOne, groupTwo],
+      toDelete: false,
+    });
+
+    fieldMap = await getKnownPersonFieldMap(
+      client,
+      { appId, appOptions },
+      true
+    );
+    groupOneKey = fieldMap[groupOneField];
+    expect(groupOneKey).toBeTruthy();
+
+    const filterOne = await findFilter(groupOneField);
+    expect(filterOne).toBeTruthy();
+
+    groupTwoKey = fieldMap[groupTwoField];
+    expect(groupTwoKey).toBeTruthy();
+
+    const filterTwo = await findFilter(groupTwoField);
+    expect(filterTwo).toBeTruthy();
+
+    const { data } = await client.PersonsController.getDetailsOfAPerson(
+      personId
+    );
+    expect(data[groupOneKey]).toBeTruthy();
+    expect(data[groupTwoKey]).toBeTruthy();
+  });
+
+  test("can remove group membership", async () => {
+    await runExport({
+      oldProfileProperties: { Email: email1, Name: "Johnny Doe" },
+      newProfileProperties: { Email: email1, Name: "Johnny Doe" },
+      oldGroups: [groupOne, groupTwo],
+      newGroups: [groupOne],
+      toDelete: false,
+    });
+
+    const { data } = await client.PersonsController.getDetailsOfAPerson(
+      personId
+    );
+    expect(data[groupOneKey]).toBeTruthy();
+    expect(data[groupTwoKey]).toBeFalsy();
+  });
+
+  test("can change email", async () => {
+    await runExport({
+      oldProfileProperties: { Email: email1, Name: "Johnny Doe" },
+      newProfileProperties: {
+        Email: email2,
+        Name: "Johnny Doe",
+      },
+      oldGroups: [],
+      newGroups: [],
+      toDelete: false,
+    });
+
+    const { data } = await client.PersonsController.getDetailsOfAPerson(
+      personId
+    );
+    expect(data.email).toHaveLength(1);
+    expect(data.email[0].value).toBe(email2);
+  });
+
+  test("can delete a Person", async () => {
+    await runExport({
+      oldProfileProperties: {
+        Email: email2,
+        Name: "Johnny Doe",
+      },
+      newProfileProperties: {
+        Email: email2,
+        Name: "Johnny Doe",
+      },
+      oldGroups: [],
+      newGroups: [],
+      toDelete: true,
+    });
+
+    const { data } = await client.PersonsController.getDetailsOfAPerson(
+      personId
+    );
+    expect(data.active_flag).toBe(false);
+  });
+
+  test("can add a Person passing a nonexistent email on the oldProfileProperties", async () => {
+    await runExport({
+      oldProfileProperties: { Email: nonexistentEmail, Name: "Bobby" },
+      newProfileProperties: { Email: email3, Name: "Bobby" },
+      oldGroups: [],
+      newGroups: [],
+      toDelete: false,
+    });
+
+    const newPersonId = await client.EnhancedPersonsController.findPersonIdByEmail(
+      email3
+    );
+    expect(newPersonId).toBeTruthy();
+
+    const { data } = await client.PersonsController.getDetailsOfAPerson(
+      newPersonId
+    );
+    expect(data.name).toBe("Bobby");
+    expect(data.email).toHaveLength(1);
+    expect(data.email[0].value).toBe(email3);
+  });
+});
```

Comment:
Please also add a test that confirms that a new Profile that is in at least one Group can be created from scratch.  Sometimes the destination APIs get wonky, have a cache delay, or something dumb.